modules/common/src/main/scala/surge/internal/health/windows/stream/WindowingHealthSignalStream.scala

```diff
@@ -2,41 +2,55 @@
 
 package surge.internal.health.windows.stream
 
-import java.util.concurrent.ArrayBlockingQueue
-
 import akka.Done
-import akka.actor.ActorRef
+import akka.actor.{ ActorRef, ActorSystem }
+import akka.stream.Materializer
+import akka.stream.scaladsl.{ Sink, Source, SourceQueueWithComplete }
 import surge.health.{ HealthSignalStream, SignalHandler }
 import surge.health.domain.HealthSignal
-import surge.health.matchers.SignalPatternMatcher
-import surge.health.windows.Window
+import surge.utils.BooleanWrapper
 
 import scala.util.Try
 
 trait StreamHandle {
-  def release(): Unit
+  private val running: BooleanWrapper = new BooleanWrapper().asTrue()
```

Comment:
Hmm let's just make this a `private var` for now and avoid the `BooleanWrapper` altogether. It's still mutable under the hood using a mutable map so it doesn't really buy us much and makes it a bit more abstract

Reply:
point taken.  again trying to avoid the `var shaming`..

Reply:
done